flang/lib/Lower/PFTBuilder.cpp

```diff
@@ -1041,6 +1041,9 @@ struct SymbolDependenceDepth {
 
 void Fortran::lower::pft::FunctionLikeUnit::processSymbolTable(
     const semantics::Scope &scope) {
+  // TODO: handle equivalence and common blocks
+  CHECK(scope.equivalenceSets().empty() &&
+        "equivalence not yet handled in lowering");
```

Comment:
Will using CHECK be a political issue?

Reply:
I think this plays out to allow CHECK for functions with side-effects.

Reply:
CHECK is a flang front-end thing that is used nowhere else. I'd rather not be dragged into the disagreement about asserts, etc. with lowering or FIR.

Reply:
You can find CHECK in other subprojects, e.g. ./lld/include/lld/Common/ErrorHandler.h, which is the most comparable.

Reply:
A different macro with the same spelling isn't exactly what I meant. ;)

Reply:
Changed to `llvm::errs<< "message " + procedure name` and `exit(1)`. It has the advantage of failing faster (skips the core dump, which is not very useful for a TODO).